src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -1152,64 +1159,70 @@ private Future<List<String>> getLocateUserFields(List<String> fieldAliasList, Ma
    * @return
    */
   private Future<User> locateUserByAlias(List<String> fieldAliasList, Identifier entity,
-                                                       Map<String, String> okapiHeaders, String errorKey) {
-    Promise<User> asyncResult = Promise.promise();
-    getLocateUserFields(fieldAliasList, okapiHeaders).onComplete(locateUserFieldsAR -> {
-      if (!locateUserFieldsAR.succeeded()) {
-        asyncResult.fail(locateUserFieldsAR.cause());
-        return;
-      }
-      String query = buildQuery(locateUserFieldsAR.result(), entity.getId());
-
-      String tenant = okapiHeaders.get(OKAPI_TENANT_HEADER);
-      String okapiURL = okapiHeaders.get(OKAPI_URL_HEADER);
-
-      HttpClientInterface client = HttpClientFactory.getHttpClient(okapiURL, tenant);
-
-      okapiHeaders.remove(OKAPI_URL_HEADER);
+                                         Map<String, String> okapiHeaders, String errorKey) {
+    return getLocateUserFields(fieldAliasList, okapiHeaders)
+      .compose(locateUserFieldsAR -> crossTenantUserService.locateCrossTenantUser(entity.getId(), okapiHeaders, errorKey)
+        .compose(user -> {
+          if (user == null) {
+            return locateUser(locateUserFieldsAR, entity, okapiHeaders, errorKey);
+          }
+          return Future.succeededFuture(user);
+          }));
+  }
 
-      try {
-        String userUrl = new StringBuilder("/users?").append("query=").append(URLEncoder.encode(query, "UTF-8")).append("&").append("offset=0&limit=2").toString();
+  /**
+   * Searching user by query
+   * @param locateUserFieldsAR a list of user fields to use for search
+   * @param entity - an identity with a value
+   * @param okapiHeaders request headers
+   * @return User
+   */
+  private Future<User> locateUser(List<String> locateUserFieldsAR, Identifier entity, Map<String, String> okapiHeaders, String errorKey) {
```

Comment:
`locateUserFieldsAR` seems like strange variable name. Is `AR` intended to mean "array"? If so I think usually the type is not important. Just name the variable as `locateUserFields` instead.

Reply:
renamed to locateUserFields